src/ports/xtensa_esp32/hardware_timer.rs

```diff
@@ -1,33 +1,14 @@
+use crate::ports::xtensa_esp32::peripherals::{
+    init_peripherals, PERIFERALS_RADIO_CLK, PERIFERALS_RNG, PERIFERALS_WIFI, TIMER00, TIMER10,
+};
 use core::sync::atomic::{AtomicBool, Ordering};
 use core::time::Duration;
 use esp_hal::timer::timg::{Timer, Timer0, TimerGroup};
 use esp_hal::{peripherals::*, prelude::*};
 
-// TODO: initialize peripherals in separate mod
-pub static mut TIMER00: Option<Timer<Timer0<TIMG0>, esp_hal::Blocking>> = None;
-pub static mut TIMER10: Option<Timer<Timer0<TIMG1>, esp_hal::Blocking>> = None;
-pub static mut PERIFERALS_RNG: Option<RNG> = None;
-pub static mut PERIFERALS_RADIO_CLK: Option<RADIO_CLK> = None;
-pub static mut PERIFERALS_WIFI: Option<WIFI> = None;
-
 static TIMER_BUSY: AtomicBool = AtomicBool::new(false);
-
-/// Esp32 hardware timer setup.
 pub fn setup_hardware_timer() {
-    let peripherals = esp_hal::init(esp_hal::Config::default());
-    let timer_group0 = TimerGroup::new(peripherals.TIMG0);
-    let timer_group1 = TimerGroup::new(peripherals.TIMG1);
-
-    let timer00 = timer_group0.timer0;
-    let timer10 = timer_group1.timer0;
-
-    unsafe {
-        TIMER00 = Some(timer00);
-        TIMER10 = Some(timer10);
-        PERIFERALS_RNG = Some(peripherals.RNG);
-        PERIFERALS_RADIO_CLK = Some(peripherals.RADIO_CLK);
-        PERIFERALS_WIFI = Some(peripherals.WIFI);
-    }
+    init_peripherals();
```

Comment:
Peripherals shouldn't be initialized in timer initialization. They should be initialized in lib.rs in init_system